### PR TITLE
Improve upgrade docs

### LINKED ISSUE #3512: dead link
On https://knative.dev/docs/install/upgrade-installation/ there is a link to the operator upgrade instructions, which does not exist: https://knative.dev/docs/install/upgrade-installation-with-operator.md

### PR DESCRIPTION
**Pull request by abrennan89**
Fixes #3512 

## Proposed Changes 

- Moved upgrade docs to their own section
- General review / editing
- Clean up versions, dead links

### CODE REVIEW

**Review by abrennan89 — COMMENTED** [review]
/approve
/hold

File: docs/upgrade/upgrade-installation-with-operator.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+```yaml
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  version: "0.22"
+```
```
Comment (abrennan89) [inline docs/upgrade/upgrade-installation-with-operator.md:28]:
```suggestion
```yaml
kubectl apply -f - <<EOF
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeServing
metadata:
  name: knative-serving
  namespace: knative-serving
spec:
  version: "0.22"
EOF
```
```

File: docs/upgrade/upgrade-installation-with-operator.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+```yaml
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeServing
+metadata:
+  name: knative-serving
+  namespace: knative-serving
+spec:
+  version: "0.21"
+```
```
Comment (abrennan89) [inline docs/upgrade/upgrade-installation-with-operator.md:108]:
```suggestion
```yaml
kubectl apply -f - <<EOF
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeServing
metadata:
  name: knative-serving
  namespace: knative-serving
spec:
  version: "0.21"
EOF
```
```

File: docs/upgrade/upgrade-installation-with-operator.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+```yaml
+apiVersion: operator.knative.dev/v1alpha1
+kind: KnativeEventing
+metadata:
+  name: knative-eventing
+  namespace: knative-eventing
+spec:
+  version: "0.21"
+```
```
Comment (abrennan89) [inline docs/upgrade/upgrade-installation-with-operator.md:121]:
```suggestion
```yaml
kubectl apply -f - <<EOF
apiVersion: operator.knative.dev/v1alpha1
kind: KnativeEventing
metadata:
  name: knative-eventing
  namespace: knative-eventing
spec:
  version: "0.21"
EOF
```
```

**Review by csantanapr — APPROVED** [review]
LGTM

**Comment by abrennan89** [comment]
OOF, I'll do a manual CP

**Pull request merged**